tests/test_reweighting.py

```diff
@@ -0,0 +1,392 @@
+    vec_u_1_fxn = lambda xs, params: vec_u(xs, 1, params)
+
+    estimate_delta_f = construct_endpoint_reweighting_estimator(
+        samples_0, samples_1, vec_u_0_fxn, vec_u_1_fxn, ref_params, ref_delta_f
+    )
+    analytical_delta_f = lambda params: reduced_free_energy(1.0, params) - reduced_free_energy(0.0, params)
+
+    assert_estimator_accurate(jit(estimate_delta_f), analytical_delta_f, ref_params, n_random_trials=10, atol=5e-3)
+
+
+def test_mixture_reweighting_1d():
+    """using a variety of free energy estimates (MBAR, TI, analytical) to obtain reference mixture weights,
+    assert that mixture reweighting estimator of delta_f(params), grad(delta_f)(params) is accurate
+    on tractable 1D system"""
+    onp.random.seed(2022)
+
+    u_fxn, normalized_u_fxn, sample, reduced_free_energy = make_gaussian_testsystem()
+
+    ref_params = np.ones(2)
+    n_windows = 10
+    lambdas = np.linspace(0, 1, n_windows)
+
+    n_samples_per_window = int(1e5)
+    N_k = [n_samples_per_window] * n_windows
+    n_samples_total = sum(N_k)
+
+    trajs = [sample(lam, ref_params, n_samples_per_window) for lam in lambdas]
+    xs = np.vstack(trajs).flatten()
+    u_kn = onp.zeros((n_windows, n_samples_total))
+    vec_u = vmap(u_fxn, in_axes=(0, None, None))
+
+    # TODO [generality] : change signature to be (lam, params) instead of (xs, lam, params)?
+    vec_u_0_fxn = lambda xs, params: vec_u(xs, 0.0, params)
+    vec_u_1_fxn = lambda xs, params: vec_u(xs, 1.0, params)
+
+    for k in range(n_windows):
+        u_kn[k] = vec_u(xs, lambdas[k], ref_params)
+
+    # f_k estimates can come from any source, as long as they're accurate
+
+    # using analytical f_k at ref_params
+    f_k_exact = np.array([reduced_free_energy(lam, ref_params) for lam in lambdas])
+    u_mix_exact = interpret_as_mixture_potential(u_kn, f_k_exact, N_k)
+
+    # various approximations to f_k at ref_params
+
+    # MBAR
+    mbar = pymbar.MBAR(u_kn, N_k=N_k)
+    f_k_mbar = mbar.f_k
+    u_mix_mbar = interpret_as_mixture_potential(u_kn, f_k_mbar, N_k)
+
+    # TI
+    vec_du_dl = vmap(grad(u_fxn, 1), (0, None, None))
+    mean_du_dls = np.array([np.mean(vec_du_dl(traj, lam, ref_params)) for (traj, lam) in zip(trajs, lambdas)])
+    f_k_ti = np.array([np.trapz(mean_du_dls[:k], lambdas[:k]) for k in range(n_windows)])
+    u_mix_ti = interpret_as_mixture_potential(u_kn, f_k_ti, N_k)
+
+    # TODO [overkill] : BAR
+    # ...
+    # u_mix_bar = ...
+
+    # TODO [overkill] : SMC
+    # ...
+    # u_mix_smc = ...
+
+    u_mixes = dict(exact_f_k=u_mix_exact, mbar_f_k=u_mix_mbar, ti_f_k=u_mix_ti)
+    analytical_delta_f = lambda params: reduced_free_energy(1.0, params) - reduced_free_energy(0.0, params)
+
+    for condition in u_mixes:
+        u_mix = u_mixes[condition]
+
+        # TODO [sign convention]: change signature to accept u_mix rather than log_weights?
+        estimate_delta_f = jit(construct_mixture_reweighting_estimator(xs, -u_mix, vec_u_0_fxn, vec_u_1_fxn))
```

Comment:
Inconsistent sign convention -- waffling between "reduced potential"s and "reduced_free_energies" everywhere vs. "log_pdf"s and "log_normalizing_constant"s everywhere...